`gh agent-task list`: implement `-w`/`--web` to open global agent tasks page in a browser

BagToad opened the pull request:
> [!WARNING]
> This is stacked onto #11620. Please merge #11620 first! 

Introduces a `--web` flag to the agent-task list command, allowing users to open the agent tasks page in their browser.

BagToad committed:
Add --web flag to agent-task list command

Introduces a --web flag to the agent-task list command, allowing users to open the agent tasks page in their browser. Updates tests to cover the new flag and browser interaction.

BagToad committed:
Merge branch 'kw/1003-gh-agent-task-list-respects--l--limit' into kw/1004-gh-agent-task-list-respects--w--web

babakks commented on pkg/cmd/agent-task/list/list.go:
```diff
@@ -77,6 +83,14 @@ func NewCmdList(f *cmdutil.Factory, runF func(*ListOptions) error) *cobra.Comman
 
 func listRun(opts *ListOptions) error {
+	if opts.Web {
+		const webURL = "https://github.com/copilot/agents"
```
Maybe with a comment that says there's no repo-specific URL so we ignore the base repo.

```suggestion
		// Currently the web GUI does not support filtering based on repo, so we
		// just open the agents dashboard with no args.
		const webURL = "https://github.com/copilot/agents"
```

BagToad replied:
Good idea. Added in 93f7e9847f33d22451b4e3d3ba0b66b781309819

babakks commented on pkg/cmd/agent-task/list/list_test.go:
```diff
@@ -168,6 +181,14 @@ func Test_listRun(t *testing.T) {
 			r6          #306          OWNER/REPO  mystery        about 6 hours ago
 			`),
 		},
+		{
+			name:           "web mode",
+			tty:            true,
+			web:            true,
+			wantOut:        "",
+			wantStderr:     "Opening https://github.com/copilot/agents in your browser.\n",
+			wantBrowserURL: "https://github.com/copilot/agents",
+		},
```
Maybe also another test case that asserts the URL is the same thing even if `opts.BaseRepo` is set.

BagToad replied:
Done in c4c0ddc8c265c82e077d21d05ad100cc2dbe7406

babakks commented on pkg/cmd/agent-task/list/list_test.go:
```diff
@@ -181,16 +202,28 @@ func Test_listRun(t *testing.T) {
+				Browser: br,
+				CapiClient: func() (*capi.CAPIClient, error) {
+					if tt.web {
+						panic("CapiClient should not be invoked when --web is set")
```
How about this? It'd be a less noisy test log this way.

```suggestion
						require.FailNow(t, "CapiClient should not be invoked when --web is set")
```

BagToad replied:
Nice, I didn't know about that, thanks!

c5f7be9adbb77c75eae7f652a9f3cdbb9354f13c

babakks commented on pkg/cmd/agent-task/list/list_test.go:
```diff
@@ -207,6 +240,14 @@ func Test_listRun(t *testing.T) {
+				require.Equal(t, tt.wantStderr, stderr.String())
+			} else {
+				require.Equal(t, "", stderr.String())
+			}
```
The `if` is redundant and can be simplified to just this:

```go
require.Equal(t, tt.wantStderr, stderr.String())
```

BagToad replied:
Fixed in 163eb7e7c88c0049d734fb28fe732137f6cf4060

babakks reviewed (approved):
Looks good! Again approving to unblock since we're doing stacked PRs.

BagToad committed:
Simplify BaseRepo assignment in list command

Removes unnecessary nil check for Factory before assigning BaseRepo in the agent-task list command. This streamlines the code since Factory is always expected to be non-nil.

BagToad committed:
Simplify time calculation in listRun test

Replaces explicit duration parsing with direct multiplication for calculating a timestamp 6 hours ago in Test_listRun, improving code clarity.

BagToad committed:
Add comment explaining error handling in listRun

Added a comment to clarify why the error from opts.BaseRepo() is ignored in listRun. This provides context for handling cases when the current working directory is not a repo and --repo is not set.

BagToad committed:
Return NoResultsError when no agent tasks found

Replaces plain output with a NoResultsError in listRun when no agent tasks are found. Updates related tests to expect the error instead of output, improving error handling consistency.

BagToad committed:
Add test case for negative limit in list command

Introduces a test to verify that passing a negative value to the --limit flag in the agent-task list command returns the expected error message.

BagToad committed:
Add comment on web dashboard filtering limitation

Added a comment explaining that the web GUI does not currently support filtering by repo, so the agents dashboard is opened without arguments. Notes future improvement if repo filtering becomes available.

BagToad committed:
Add test for web mode with repo flag in listRun

Added a test case to ensure that web mode in listRun uses the global URL even when the --repo flag is set. This improves coverage for scenarios where both web mode and repo are specified.

BagToad committed:
Replace panic with require.FailNow in test

Updated the test to use require.FailNow instead of panic when CapiClient is called with --web, improving test failure reporting and consistency.
